sdk/tests/conformance2/textures/misc/tex-image-with-bad-args-from-dom-elements.html

```
@@ -0,0 +1,147 @@
+    wtu.glErrorShouldBe(gl, [gl.INVALID_VALUE, gl.INVALID_ENUM, gl.INVALID_OPERATION], "TexImage2D taking RGB10_A2/RGBA/UNSIGNED_INT_2_10_10_10_REV should fail");
+    gl.deleteTexture(tex);
+}
+
+var createCanvas2DContext = function(width, height) {
+    var canvas = document.createElement("canvas");
+    canvas.width = width;
+    canvas.height = height;
+    var ctx = canvas.getContext("2d");
+    ctx.fillRect(0, 0, width, height);
+    return ctx;
+};
+
+var testImage = function(test) {
+  debug("HTMLImageElement");
+  var img = wtu.makeImage(test.src, function() {
+    doTexImage(img);
+    setTimeout(runNextTest, 0);
+  });
+}
+
+var testVideo = function(test) {
+  debug("HTMLVideoElement (" + test.videoType + ")");
+  var video = wtu.makeVideo(test.src);
+  if(!video.canPlayType(test.videoType).replace(/no/, '')) {
+    debug(test.videoType + " unsupported");
+    setTimeout(runNextTest, 0);
+    return;
+  }
+  wtu.startPlayingAndWaitForVideo(video, function() {
+    doTexImage(video);
+    setTimeout(runNextTest, 0);
+  });
+}
+
+var testCanvas = function(test) {
+  var ctx = createCanvas2DContext(test.width, test.height);
+  debug("HTMLCanvasElement");
+  doTexImage(ctx.canvas);
+  setTimeout(runNextTest, 0);
+}
+
+var testImageData = function(test) {
+  var ctx = createCanvas2DContext(test.width, test.height);
+  var imageData = ctx.getImageData(0, 0, test.width, test.height);
+  debug("ImageData");
+  doTexImage(imageData);
+  setTimeout(runNextTest, 0);
+}
+
+var testImageBitmap = function(test) {
+  debug("ImageBitmap");
+  if(!window.createImageBitmap || !window.ImageBitmap) {
+    debug("ImageBitmap isn't supported");
+    setTimeout(runNextTest, 0);
+    return;
+  }
+  var ctx = createCanvas2DContext(test.width, test.height);
+  createImageBitmap(ctx.canvas, 0, 0, test.width, test.height).then(function(imageBitmap) {
+    doTexImage(imageBitmap);
+    setTimeout(runNextTest, 0);
+  }, function() {
+    debug("createImageBitmap was rejected");
```

Should failure to create the ImageBitmap cause the test to fail rather than just a warning? CC @xidachen .